### PR TITLE
add Home and Rules page

### PR DESCRIPTION
**Pull request by hsnice16**
### This PR will add code for the design of the Home and Rules page

In this PR, **446 additions 49 deletions** line of code
- some files contain only 5 to 6 lines of code
- some files are for images, so you can avoid reviewing them

### Please review this PR and give your feedback
---
#### [Live URL](https://react-marvels-quiz-git-page-home-rules-hsnice16.vercel.app/)

image

image

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -109,12 +85,7 @@ li:not(.option-li, .answer-li) {
...
 
-/** Overriding **/
-
-.card {
-  overflow: hidden;
-  width: 33rem;
-}
+/* overriding */
 
 .stacked-list > li {
```
Comment (Naman-Saxena1) [inline src/App.css:90]: .stacked-list > list
Avoid complex CSS selectors if possible using some other alternative

File: src/custom-hooks/useScrollToTop.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ */
+const useScrollToTop = (scrollWhenChange = "") => {
+  useEffect(() => {
+    // eslint-disable-next-line no-restricted-globals
+    scrollTo({
+      top: 0,
+    });
+  }, [scrollWhenChange]);
+};
+
```
Comment (Naman-Saxena1) [inline src/custom-hooks/useScrollToTop.js:17]: Appreciation: I liked how you properly segregated files according to functionality.

**Review by Naman-Saxena1 — APPROVED** [review]
Code quality is great, properly separated files, descriptive variable names and other good practices are followed 👍 
I also really like UI of app
I have 1 small suggestion, Captain Marvel is getting out of card, maybe crop/edit it to show her face in card.
Capture

File: src/App.css (hunk 1 above)
Comment (hsnice16) [reply src/App.css:90]: Thanks, Naman for pointing this out, as It was only level that's why I didn't find any alternative. But will keep this point in mind.

**Comment by hsnice16** [comment]
> Code quality is great, properly separated files, descriptive variable names and other good practices are followed 👍 I also really like UI of app I have 1 small suggestion, Captain Marvel is getting out of card, maybe crop/edit it to show her face in card. Capture

Thanks Naman for this suggestion, I had tried using `object-position` to position the image differently but this was the only which was looking nice. 
But will look if I can find something else.

File: src/components/button/button.css [hunk 3]
```diff
@@ -0,0 +1,21 @@
+.btn-back-home,
+.btn-game-begin {
+  background-color: var(--COLOR-PRIMARY);
+  border: none;
+  border-radius: 0.4rem;
```
Comment (prachi-sahani) [inline src/components/button/button.css:5]: you can keep this in px

**Review by prachi-sahani — COMMENTED** [review]
- great UI - simple and clean 
- liked the way you are exporting assets(got to learn a new thing)
- also liked the way you have segregated the code which is making it  well structured and clean 
- good use of comments


**Suggestion:**
- remove ```{" "}``` which gets added while formatting code
- all the font-family might not be supported in all kinds of browser. Keeping the cross-browser compatibility in mind, either select fonts that are commonly used and/or provide fallback font-family(this was regarding kushan script that you have used and exported independently)
- not related to this PR, just a suggestion in general - in my opinion, too much segregation might sometimes kill the readability of the code

***Loved the way you have given PR description***

**Comment by prachi-sahani** [comment]
> > Code quality is great, properly separated files, descriptive variable names and other good practices are followed 👍 I also really like UI of app I have 1 small suggestion, Captain Marvel is getting out of card, maybe crop/edit it to show her face in card. Capture
> 
> Thanks Naman for this suggestion, I had tried using `object-position` to position the image differently but this was the only which was looking nice. But will look if I can find something else.

You can try using ```object-fit``` property (not sure if that will help)

**Comment by hsnice16** [comment]
> > > Code quality is great, properly separated files, descriptive variable names and other good practices are followed 👍 I also really like UI of app I have 1 small suggestion, Captain Marvel is getting out of card, maybe crop/edit it to show her face in card. Capture
> > 
> > 
> > Thanks Naman for this suggestion, I had tried using `object-position` to position the image differently but this was the only which was looking nice. But will look if I can find something else.
> 
> You can try using `object-fit` property (not sure if that will help)

I have used `object-fit`. It is in my CL. But thanks for suggesting.

**Comment by hsnice16** [comment]
> * too much segregation might sometimes kill the readability of the code

Suggestion:
- Will check for the `{" "}` in code and will remove it. 
- I am using google fonts so I think it must be compatible with cross-browsers. And also there is a fallback value present
```css
font-family: "Kaushan Script", cursive;
```

**Comment by hsnice16** [comment]
Thank you for reviewing,

**Comment by prachi-sahani** [comment]
> > * too much segregation might sometimes kill the readability of the code
> 
> Suggestion:
> 
> * Will check for the `{" "}` in code and will remove it.
> * I am using google fonts so I think it must be compatible with cross-browsers. And also there is a fallback value present
> 
> ```css
> font-family: "Kaushan Script", cursive;
> ```

ohh yes, then it is fine 🙂

**Pull request merged**